doc/devdocs/locks.rst

```diff
@@ -0,0 +1,128 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free
+(generally by addressing the 4th Coffman condition: circular wait).
```

Comment:
I'd wager that the majority of Julia users, and a fair number of its developers, have never heard of that reference. Title, author, year.